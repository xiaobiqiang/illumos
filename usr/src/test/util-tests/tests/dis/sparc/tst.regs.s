/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2016 Joyent, Inc.
 */

/*
 * Test basic register naming
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	add %g0, %g1, %g2
	add %g1, %g2, %g3
	add %g2, %g3, %g4
	add %g3, %g4, %g5
	add %g4, %g5, %g6
	add %g5, %g6, %g7
	add %g6, %g7, %o0
	add %g7, %o0, %o1
	add %o0, %o1, %o2
	add %o1, %o2, %o3
	add %o2, %o3, %o4
	add %o3, %o4, %o5
	add %o4, %o5, %sp
	add %o5, %sp, %o7
	add %sp, %o7, %l0
	add %o7, %l0, %l1
	add %l0, %l1, %l2
	add %l1, %l2, %l3
	add %l2, %l3, %l4
	add %l3, %l4, %l5
	add %l4, %l5, %l6
	add %l5, %l6, %l7
	add %l6, %l7, %i0
	add %l7, %i0, %i1
	add %i0, %i1, %i2
	add %i1, %i2, %i3
	add %i2, %i3, %i4
	add %i3, %i4, %i5
	add %i4, %i5, %fp
	add %i5, %fp, %i7
.size libdis_test, [.-libdis_test]
