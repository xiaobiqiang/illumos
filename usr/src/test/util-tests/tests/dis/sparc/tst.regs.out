    libdis_test:      84 00 00 01  add       %g0, %g1, %g2
    libdis_test+0x4:  86 00 40 02  add       %g1, %g2, %g3
    libdis_test+0x8:  88 00 80 03  add       %g2, %g3, %g4
    libdis_test+0xc:  8a 00 c0 04  add       %g3, %g4, %g5
    libdis_test+0x10: 8c 01 00 05  add       %g4, %g5, %g6
    libdis_test+0x14: 8e 01 40 06  add       %g5, %g6, %g7
    libdis_test+0x18: 90 01 80 07  add       %g6, %g7, %o0
    libdis_test+0x1c: 92 01 c0 08  add       %g7, %o0, %o1
    libdis_test+0x20: 94 02 00 09  add       %o0, %o1, %o2
    libdis_test+0x24: 96 02 40 0a  add       %o1, %o2, %o3
    libdis_test+0x28: 98 02 80 0b  add       %o2, %o3, %o4
    libdis_test+0x2c: 9a 02 c0 0c  add       %o3, %o4, %o5
    libdis_test+0x30: 9c 03 00 0d  add       %o4, %o5, %sp
    libdis_test+0x34: 9e 03 40 0e  add       %o5, %sp, %o7
    libdis_test+0x38: a0 03 80 0f  add       %sp, %o7, %l0
    libdis_test+0x3c: a2 03 c0 10  add       %o7, %l0, %l1
    libdis_test+0x40: a4 04 00 11  add       %l0, %l1, %l2
    libdis_test+0x44: a6 04 40 12  add       %l1, %l2, %l3
    libdis_test+0x48: a8 04 80 13  add       %l2, %l3, %l4
    libdis_test+0x4c: aa 04 c0 14  add       %l3, %l4, %l5
    libdis_test+0x50: ac 05 00 15  add       %l4, %l5, %l6
    libdis_test+0x54: ae 05 40 16  add       %l5, %l6, %l7
    libdis_test+0x58: b0 05 80 17  add       %l6, %l7, %i0
    libdis_test+0x5c: b2 05 c0 18  add       %l7, %i0, %i1
    libdis_test+0x60: b4 06 00 19  add       %i0, %i1, %i2
    libdis_test+0x64: b6 06 40 1a  add       %i1, %i2, %i3
    libdis_test+0x68: b8 06 80 1b  add       %i2, %i3, %i4
    libdis_test+0x6c: ba 06 c0 1c  add       %i3, %i4, %i5
    libdis_test+0x70: bc 07 00 1d  add       %i4, %i5, %fp
    libdis_test+0x74: be 07 40 1e  add       %i5, %fp, %i7
